src/libs/structs/state.rs

```diff
@@ -72,6 +74,31 @@ pub struct StrataState<BackendData: Backend + 'static> {
 	pub pointer_location: Point<f64, Logical>,
 }
 
+pub struct GlobalState<BackendData: Backend + 'static> {
+	inner: OnceCell<Mutex<StrataState<BackendData>>>,
+}
+
```

Comment:
```suggestion

	impl<BackendData> Deref for GlobalState<BackendData> {
		type Target = StrataState<BackendData>;
		
		fn deref(&self) -> Target {
			self.get().expect("uninitialized")
		}
	}

```
This will allow you to call `GLOBAL_STATE.lock()` and use it directly without boilerplate.